packages/discovery-provider/src/api/v1/tracks.py

```diff
@@ -465,49 +461,38 @@ def get(self, track_id):
         """
         request_args = stream_parser.parse_args()
         is_preview = request_args.get("preview")
-        filename = request_args.get("filename")
         decoded_id = decode_with_abort(track_id, ns)
-        info = get_track_stream_info(decoded_id)
+        info = get_track_access_info(decoded_id)
 
         track = info.get("track")
-        cid = track.get("track_cid")
-        if is_preview:
-            cid = track.get("preview_cid")
-        elif filename:
-            cid = track.get("orig_file_cid")
 
-        if not track or not cid:
+        if not track:
             logger.error(
-                f"tracks.py | stream | Track with id {track_id} may not exist or has no {'preview' if is_preview else 'track'}_cid. Please investigate."
+                f"tracks.py | stream | Track with id {track_id} may not exist. Please investigate."
             )
             abort_not_found(track_id, ns)
 
-        cid = cid.strip()
         redis = redis_connection.get_redis()
 
         # signature for the track to be included as a query param in the redirect to CN
-        signature = get_track_stream_signature(
+        stream_signature = get_track_stream_signature(
             {
                 "track": track,
-                "cid": cid,
                 "is_preview": is_preview,
-                "is_download": filename is not None,
                 "user_data": request_args.get("user_data"),
                 "user_signature": request_args.get("user_signature"),
-                "premium_content_signature": request_args.get(
-                    "premium_content_signature"
-                ),
+                "stream_signature": request_args.get("stream_signature"),
             }
         )
-        if not signature:
+        if not stream_signature:
             abort_not_found(track_id, ns)
 
+        signature = stream_signature["signature"]
+        cid = stream_signature["cid"]
         params = {"signature": json.dumps(signature)}
         skip_play_count = request_args.get("skip_play_count", False)
         if skip_play_count:
             params["skip_play_count"] = skip_play_count
-        if filename:
-            params["filename"] = filename
```

Comment:
Maybe to preserve the existing functionality, 302 redirect to the download endpoint?

Reply:
done, in subsequent pr :)